src/components/common/FrequencyAdjuster.js

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withTranslation } from 'react-i18next';
+import Typography from '@material-ui/core/Typography';
+import Slider from '@material-ui/core/Slider';
+import IconButton from '@material-ui/core/IconButton';
+import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
+import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';
+import { withStyles } from '@material-ui/core/styles';
+import { adjustFrequency } from '../../actions';
+import {
+  FREQUENCY_ADJUSTMENT_FACTOR_DEFAULT,
+  FREQUENCY_ADJUSTMENT_FACTOR_MIN,
+  FREQUENCY_ADJUSTMENT_FACTOR_MAX,
+  FREQUENCY_ADJUSTMENT_FACTOR_STEP,
+} from '../../config/constants';
+
+const styles = (theme) => ({
+  container: {
+    display: 'flex',
+    alignItems: 'center',
+    margin: theme.spacing(2, 0),
+  },
+  typography: {
+    marginRight: theme.spacing(2),
+  },
+  button: {
+    borderRadius: '50%',
+  },
+});
+
+class FrequencyAdjuster extends Component {
+  static propTypes = {
+    t: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      container: PropTypes.string.isRequired,
+      typography: PropTypes.string.isRequired,
+      button: PropTypes.string.isRequired,
+    }).isRequired,
+    dispatchAdjustFrequency: PropTypes.func.isRequired,
+    shouldOscillate: PropTypes.bool.isRequired,
+  };
+
+  state = {
+    sliderValue: FREQUENCY_ADJUSTMENT_FACTOR_DEFAULT,
+  };
+
+  increaseFrequency = () => {
+    const { shouldOscillate } = this.props;
+    if (shouldOscillate) {
+      this.setState(
+        (prevState) => {
+          const newState =
+            prevState.sliderValue + FREQUENCY_ADJUSTMENT_FACTOR_STEP >=
```

Comment:
Could achieve this with `Math.min(prevState.slideValue + FREQUENCY_ADJUSTMENT_FACTOR_STEP, FREQUENCY_ADJUSTMENT_FACTOR_MAX)`

Reply:
Absolutely right, but to keep consistency between `decreaseFrequency` and `increaseFrequency` functions, will leave as is